src/main/java/io/github/anthorx/parquet/sql/write/JDBCWriter.java

```diff
@@ -88,23 +87,22 @@ public void write() throws IOException, SQLException {
     }
   }
 
-  private List<RecordField> extractFieldValues(List<Type> fields, Record record) {
+  private List<RecordField<?>> extractFieldValues(List<Type> fields, Record record) {
     return fields
         .stream()
         .reduce(new ArrayList<>(), (currentValues, currentField) -> {
-          RecordField result = record
+          RecordField<?> result = record
               .getField(currentField.getName())
-              .orElse(createNullRecordField(currentField));
+              .orElseGet(() -> createNullRecordField(currentField));
           currentValues.add(result);
           return currentValues;
         }, (before, after) -> after);
   }
 
-  private RecordField createNullRecordField(Type currentField) {
-    RecordField<Object> recordField = new RecordField<>(currentField.getName(), null);
-    BiConsumer<ReadRecordConsumer, Object> objectRecordConsumer = ReadRecordConsumer::setObject;
-    recordField.addReadConsumer(objectRecordConsumer);
-
+  private RecordField<?> createNullRecordField(Type currentField) {
+    Consumer<ReadRecordConsumer> setNull = NullTypeConsumer.get(currentField);
+    RecordField<?> recordField = new RecordField<>(currentField.getName(), null);
+    recordField.addReadConsumer(((readRecordConsumer, o) -> setNull.accept(readRecordConsumer)));
```

Comment:
I think this should be under the responsability of your class "NullTypeConsumer".
A function with this signature :
```java
<T> BiConsumer<ReadRecordConsumer, T> createNullReadRecordConsumer(Type currentField) {
```

Reply:
agreed